src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/ExchangeTaskMapper.kt

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.wfanet.panelmatch.client.exchangetasks
+
+import com.google.protobuf.ByteString
+import org.wfanet.measurement.api.v2alpha.ExchangeWorkflow
+import org.wfanet.panelmatch.client.storage.Storage
+
+/**
+ * Maps ExchangeWorkflow.Step to respective tasks. Retrieves necessary inputs. Executes step. Stores
+ * outputs.
+ *
+ * @param ExchangeWorkflow.Step to execute.
+ * @param input inputs needed by all [task]s.
+ * @param storage the Storage class to store the intermediary steps
+ * @param sendDebugLog function which writes logs happened during execution.
+ * @return mapped output. This is either a ByteString or null for input steps.
+ */
+class ExchangeTaskMapper {
+
+  suspend fun execute(
+    step: ExchangeWorkflow.Step,
+    input: Map<String, ByteString>,
+    storage: Storage,
+    sendDebugLog: suspend (String) -> Unit
+  ): ByteString? {
+    val inputLabels = step.getInputLabelsMap()
+    val outputLabels = step.getOutputLabelsMap()
+    val outputFieldName = requireNotNull(outputLabels["output"])
+    when (step.getStepCase()) {
+      ExchangeWorkflow.Step.StepCase.INPUT -> {
+        val inputFieldName = requireNotNull(inputLabels["input"])
+        storage.write(outputFieldName, requireNotNull(input[inputFieldName]))
+        return null
```

Comment:
Can we break inside of each step cases into private util functions? This method seems a bit long and hard to read.